core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/catalog/cache/CamelCatalogCacheManager.java

```diff
@@ -96,20 +94,66 @@ public CamelModel getDefaultCamelModel(String version) {
 		if (CamelCatalogUtils.isCamelVersionWithoutProviderSupport(version) ) {
 			coords.setArtifactId(CamelCatalogUtils.CATALOG_CAMEL_ARTIFACTID);
 		}
-		return getCachedCatalog(coords);
+		return getCachedCatalog(coords, new NullProgressMonitor());
 	}
 	
 	/**
 	 * initializes the catalog for the given coords
 	 * 
 	 * @param coordinates
+	 * @param monitor 
 	 */
-	protected void initializeCatalog(CamelCatalogCoordinates coordinates) {
+	protected void initializeCatalog(CamelCatalogCoordinates coordinates, IProgressMonitor monitor) {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, NLS.bind(Messages.initializingCamelModel, coordinates.getVersion()), 1);
 		Dependency dep = new Dependency();
 		dep.setGroupId(coordinates.getGroupId());
 		dep.setArtifactId(coordinates.getArtifactId());
 		dep.setVersion(coordinates.getVersion());
 		
 		camelModelCache.put(coordinates, CamelServiceManagerUtil.getManagerService().getCamelModel(coordinates.getVersion(), CamelCatalogUtils.getRuntimeProviderFromDependency(dep)));
+		subMonitor.setWorkRemaining(0);
 	}
+	
+	public CamelModel getCamelModelForProject(IProject project, IProgressMonitor monitor) {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, Messages.retrievingCamelModel, 3);
+		CamelCatalogCoordinates coords;
+		if (project == null) {
+			if(lastRetrievedCamelCatalog != null) {
+				return lastRetrievedCamelCatalog;
+			}
+			coords = CamelCatalogUtils.getDefaultCatalogCoordinates();
+		} else {
+			coords = CamelCatalogUtils.getCatalogCoordinatesForProject(project);
+		}
+		subMonitor.setWorkRemaining(2);
+		// initialize repos for the dep lookup
+		if (project != null) {
+			CamelServiceManagerUtil.getManagerService().updateMavenRepositoryLookup(new CamelMavenUtils().getRepositories(project), coords);
```

Comment:
Maybe we should move this line into the else case above?

Reply:
why only in the else? Seems we need it always. It was done like that before I don't see reasons to change it. Can you elaborate?

Reply:
line 125 is already handling the project != null case. So why not merging the line 131 into that case? And no, its not called always, its only called whenever the project is NOT NULL.

Reply:
ok got it, thanks